R/unnecessary_nesting_linter.R

```diff
@@ -0,0 +1,168 @@
+#' Block instances of excessive nesting
+#'
+#' Excessive nesting harms readability. Use helper functions or early returns
+#'   to reduce nesting wherever possible.
+#'
+#' @examples
+#' # will produce lints
+#' code <- "if (A) {\n  stop('A is bad!')\n} else {\n  do_good()\n}"
+#' writeLines(code)
+#' lint(
+#'   text = code,
+#'   linters = unnecessary_nesting_linter()
+#' )
+#'
+#' code <- "tryCatch(\n  {\n    foo()\n  },\n  error = identity\n)"
```

Comment:
Will `tryCatch({\n  foo()\n}, error = identity)` lint?

Reply:
yes, I'll add a test:

```r
lint("tryCatch({\n  foo()\n}, error = identity)", unnecessary_nesting_linter())
# <text>:1:10: warning: [unnecessary_nesting_linter] Reduce the nesting of this statement by removing the braces {}.
# tryCatch({
#          ^
```

Reply:
btw, styler will adjust that code with output:

```
tryCatch(
  {
    foo()
  },
  error = identity
)
```